### PR TITLE
Add `Makie` extension to Ribasim core

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/1672.

I've added some basic functionality to plot storages and levels with Makie:
basin_data

flow
(For conservative nodes I only plot the inflow)

I'm first going to work on https://github.com/Deltares/Ribasim/issues/1674 and then visualize that.

### CODE REVIEW

**Comment by visr** [comment]
Nice! Would be good to add units on the axes like in https://github.com/Deltares/Ribasim?tab=readme-ov-file#download. The water balance bar plots there may also be nice to add at some point.

File: core/ext/RibasimMakieExt.jl [hunk 1]
```diff
@@ -0,0 +1,61 @@
+module RibasimMakieExt
+using Ribasim: Ribasim, Model
+if isdefined(Base, :get_extension)
+    using Makie
+    using DataFrames
+else
+    using ..Makie
+    using ..DataFrames
+end
```
Comment (visr) [inline core/ext/RibasimMakieExt.jl:9]: Our julia compat is such that get_extension will be defined, so no need for the else block.

Could you do

```
using Makie: a, b
using DataFrames: c, d
```

Such that we explicitly see what is being used?

File: core/Project.toml [hunk 2]
```diff
@@ -40,6 +40,13 @@ TerminalLoggers = "5d786b92-1e48-4d6f-9151-6b4477ca9bed"
 TimerOutputs = "a759f4b9-e2f1-59dc-863e-4aeb61b1ea8f"
 TranscodingStreams = "3bb67fe8-82b1-5028-8e26-92a6c54297fa"
 
+[weakdeps]
+Makie = "ee78f7c6-11fb-53f2-987a-cfe4a2b5a57a"
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
```
Comment (visr) [inline core/Project.toml:45]: Can you also add compat entries for these two weakdeps, together with the other compat entries?

File: core/ext/RibasimMakieExt.jl [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        return nothing
+    end
+    label = "$(first_row.from_node_type) #$(first_row.from_node_id) → $(first_row.to_node_type) #$(first_row.to_node_id)"
+    lines!(ax, flow_data.time, flow_data.flow_rate; label)
+    return nothing
+end
+
+function Ribasim.plot_flow(model::Model)
+    f = Figure()
+    ax = Axis(f[1, 1]; xlabel = "time", ylabel = "flow rate")
```
Comment (visr) [inline core/ext/RibasimMakieExt.jl:52]:
```suggestion
    ax = Axis(f[1, 1]; xlabel = "time", ylabel = "flow rate / m³s⁻¹")
```

File: core/ext/RibasimMakieExt.jl [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    end
+
+    axislegend(ax)
+    return nothing
+end
+
+function Ribasim.plot_basin_data(model::Model)
+    f = Figure()
+    ax1 = Axis(f[1, 1]; ylabel = "level")
+    ax2 = Axis(f[2, 1]; xlabel = "time", ylabel = "storage")
```
Comment (visr) [inline core/ext/RibasimMakieExt.jl:25]:
```suggestion
    ax1 = Axis(f[1, 1]; ylabel = "level / m")
    ax2 = Axis(f[2, 1]; xlabel = "time", ylabel = "storage / m³")
```

**Review by visr — APPROVED** [review]

File: core/Project.toml (hunk 2 above)
Comment (SouthEndMusic) [reply core/Project.toml:45]: turns out we already had one for `DataFrames`

**Pull request merged**